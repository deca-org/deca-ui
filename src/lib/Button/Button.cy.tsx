import { Test } from '../Utils';
import _cyp from '../../../cypress';
import React from 'react';
import { standardColors } from '../Theme';
import Button from './Button';

interface CSSTestComposerProps {
  component: React.ReactElement;
  bg: string;
  borderColor: string;
  color: string;
  darkMode?: boolean;
}

const CSSTestComposer = ({
  component,
  bg,
  borderColor,
  color,
  darkMode = false,
}: CSSTestComposerProps) => {
  const mount = darkMode ? 'darkMount' : 'mount';
  it('base', () => {
    cy[mount](component);
    cy.get('button').should('have.css', 'cursor', 'pointer');
  });
  it('background-color', () => {
    cy[mount](component);
    cy.get('button').should('have.css', 'background-color', Test.color(bg));
  });
  it('border-color', () => {
    cy[mount](component);
    cy.get('button').should(
      'have.css',
      'border-color',
      Test.color(borderColor)
    );
  });
  it('text color', () => {
    cy[mount](component);
    cy.get('button').should('have.css', 'color', Test.color(color));
  });
  it('disabled state', () => {
    cy[mount](React.cloneElement(component, { disabled: true }));
    cy.get('button').should('have.css', 'opacity', '0.5');
    cy.get('button').should('have.css', 'cursor', 'not-allowed');
  });
};

describe('components/Button', () => {
  it('pill', () => {
    cy.mount(<Button pill>Button</Button>);
    cy.get('button').should(
      'have.css',
      'border-radius',
      Test.borderRadius('pill')
    );
  });
  describe('sizes', () => {
    it('sm', () => {
      cy.mount(<Button size="sm">Button</Button>);
      cy.get('button').should('have.css', 'paddingLeft', Test.space('2'));
      cy.get('button').should('have.css', 'paddingRight', Test.space('2'));
      cy.get('button').should('have.css', 'height', Test.size('6'));
      cy.get('button').should('have.css', 'fontSize', Test.fontSize('caption'));
    });
    it('md', () => {
      cy.mount(<Button size="md">Button</Button>);
      cy.get('button').should('have.css', 'paddingLeft', Test.space('3'));
      cy.get('button').should('have.css', 'paddingRight', Test.space('3'));
      cy.get('button').should('have.css', 'height', Test.size('9'));
      cy.get('button').should('have.css', 'fontSize', Test.fontSize('bodySm'));
    });
    it('lg', () => {
      cy.mount(<Button size="lg">Button</Button>);
      cy.get('button').should('have.css', 'paddingLeft', Test.space('4'));
      cy.get('button').should('have.css', 'paddingRight', Test.space('4'));
      cy.get('button').should('have.css', 'height', Test.size('10'));
      cy.get('button').should('have.css', 'fontSize', Test.fontSize('body'));
    });
  });
  describe('variants', () => {
    describe('solid variant', () => {
      standardColors.map((color) => {
        describe(color, () => {
          CSSTestComposer({
            component: (
              <Button variant="solid" color={color}>
                Button
              </Button>
            ),
            bg: color,
            borderColor: color,
            color: `${color}-readable`,
          });
        });
      });
    });
    describe('outlined variant', () => {
      standardColors.map((color) => {
        describe(color, () => {
          CSSTestComposer({
            component: (
              <Button variant="outlined" color={color}>
                Button
              </Button>
            ),
            bg: 'transparent',
            borderColor: color,
            color: color,
          });
        });
      });
    });
    describe('ghost variant', () => {
      standardColors.map((color) => {
        describe(color, () => {
          CSSTestComposer({
            component: (
              <Button variant="ghost" color={color}>
                Button
              </Button>
            ),
            bg: 'transparent',
            borderColor: 'transparent',
            color: color,
          });
        });
      });
    });
    describe('flat variant', () => {
      standardColors.map((color) => {
        describe(color, () => {
          CSSTestComposer({
            component: (
              <Button variant="flat" color={color}>
                Button
              </Button>
            ),
            bg: `${color}-lighten-7`,
            borderColor: `${color}-lighten-8`,
            color: color,
          });
        });
      });
    });
    describe('dark mode', () => {
      describe('flat variant', () => {
        standardColors.map((color) => {
          describe(color, () => {
            CSSTestComposer({
              component: (
                <Button variant="flat" color={color}>
                  Button
                </Button>
              ),
              bg: `${color}-lighten-5`,
              borderColor: `${color}-lighten-8`,
              color: color,
              darkMode: true,
            });
          });
        });
      });
    });
  });
});
