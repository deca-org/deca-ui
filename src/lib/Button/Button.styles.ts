import { CSS, standardColors, styled } from '@lib/Theme/stitches.config';

const compoundVariantComposer = () => {
  const colorVariants: {
    isDisabled?: boolean;
    color?: string;
    variant?: 'solid' | 'outlined' | 'ghost' | 'flat';
    isDark?: boolean;
    css: CSS;
  }[] = [];

  standardColors.map((color) => {
    colorVariants.push(
      {
        isDisabled: false,
        color: color,
        variant: 'solid',
        css: {
          color: `$${color}-readable`,
          '&:hover': {
            borderColor: `$${color}-darken-1`,
            bg: `$${color}-darken-1`,
          },
          '&:active': {
            borderColor: `$${color}-darken-2`,
            bg: `$${color}-darken-2`,
          },
        },
      },
      {
        isDisabled: true,
        color: color,
        variant: 'solid',
        css: {
          color: `$${color}-readable`,
        },
      },
      {
        isDisabled: false,
        color: color,
        variant: 'outlined',
        css: {
          bg: '$transparent',
          '&:hover': {
            borderColor: 'CurrentColor',
            color: `$${color}-darken-1`,
            bg: `$${color}-lighten-6`,
          },
          '&:active': {
            borderColor: 'CurrentColor',
            color: `$${color}-darken-2`,
            bg: `$${color}-lighten-5`,
          },
        },
      },
      {
        isDisabled: true,
        color: color,
        variant: 'outlined',
        css: {
          bg: '$transparent',
        },
      },
      {
        isDisabled: false,
        color: color,
        variant: 'ghost',
        isDark: false,
        css: {
          bg: '$transparent',
          borderColor: '$transparent',
          '&:hover': {
            color: `$${color}-darken-1`,
            borderColor: `$${color}-lighten-8`,
            bg: `$${color}-lighten-6`,
          },
          '&:active': {
            color: `$${color}-darken-2`,
            borderColor: `$${color}-lighten-8`,
            bg: `$${color}-lighten-5`,
          },
        },
      },
      {
        isDisabled: false,
        color: color,
        variant: 'ghost',
        isDark: true,
        css: {
          bg: '$transparent',
          borderColor: '$transparent',
          '&:hover': {
            color: `$${color}-darken-1`,
            borderColor: `$${color}-lighten-8`,
            bg: `$${color}-lighten-5`,
          },
          '&:active': {
            color: `$${color}-darken-2`,
            borderColor: `$${color}-lighten-8`,
            bg: `$${color}-lighten-4`,
          },
        },
      },
      {
        isDisabled: true,
        color: color,
        variant: 'ghost',
        css: {
          bg: '$transparent',
          borderColor: '$transparent',
        },
      },
      {
        isDisabled: false,
        color: color,
        variant: 'flat',
        isDark: false,
        css: {
          color: color,
          bg: `$${color}-lighten-7`,
          borderColor: `$${color}-lighten-8`,
          '&:hover': {
            color: `$${color}-darken-1`,
            borderColor: `$${color}-lighten-8`,
            bg: `$${color}-lighten-6`,
          },
          '&:active': {
            color: `$${color}-darken-2`,
            borderColor: `$${color}-lighten-8`,
            bg: `$${color}-lighten-5`,
          },
        },
      },
      {
        isDisabled: false,
        color: color,
        variant: 'flat',
        isDark: true,
        css: {
          color: color,
          bg: `$${color}-lighten-5`,
          borderColor: `$${color}-lighten-8`,
          '&:hover': {
            color: `$${color}-darken-1`,
            borderColor: `$${color}-lighten-8`,
            bg: `$${color}-lighten-4`,
          },
          '&:active': {
            color: `$${color}-darken-1`,
            borderColor: `$${color}-lighten-8`,
            bg: `$${color}-lighten-3`,
          },
        },
      },
      {
        isDisabled: true,
        color: color,
        variant: 'flat',
        css: {
          bg: `$${color}-lighten-7`,
          borderColor: `$${color}-lighten-8`,
        },
      }
    );
  });
  return colorVariants;
};

const StyledButton = styled('button', {
  flexShrink: 0,
  whiteSpace: 'nowrap',
  fontWeight: '$semibold',
  fontFamily: '$normal',
  '-webkit-font-smoothing': 'antialiased',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  boxSizing: 'border-box',
  borderWidth: '$normal',
  borderStyle: 'solid',
  cursor: 'pointer',
  outline: 0,
  transition: '$default',
  '&:focus-visible': {
    boxShadow: '$a11y',
  },
  compoundVariants: [
    ...compoundVariantComposer(),
    {
      singleIcon: true,
      size: 'sm',
      css: {
        px: '$n',
        size: '$6',
      },
    },
    {
      singleIcon: true,
      size: 'md',
      css: {
        px: '$n',
        size: '$9',
      },
    },
    {
      singleIcon: true,
      size: 'lg',
      css: {
        px: '$n',
        size: '$10',
      },
    },
  ],
  variants: {
    color: {
      primary: {
        bg: '$primary',
        color: '$primary',
        borderColor: '$primary',
      },
      secondary: {
        bg: '$secondary',
        color: '$secondary',
        borderColor: '$secondary',
      },
      success: {
        bg: '$success',
        color: '$success',
        borderColor: '$success',
      },
      warning: {
        bg: '$warning',
        color: '$warning',
        borderColor: '$warning',
      },
      error: {
        bg: '$error',
        color: '$error',
        borderColor: '$error',
      },
    },
    size: {
      sm: {
        px: '$2',
        fontSize: '$caption',
        height: '$6',
      },
      md: {
        px: '$3',
        fontSize: '$bodySm',
        height: '$9',
      },
      lg: {
        px: '$4',
        fontSize: '$body',
        height: '$10',
      },
    },
    singleIcon: {
      true: {},
      false: {},
    },
    variant: {
      solid: {},
      outlined: {},
      ghost: {},
      flat: {},
    },
    isDisabled: {
      true: {
        opacity: 0.5,
        cursor: 'not-allowed',
      },
      false: {
        opacity: 1,
      },
    },
    maxWidth: {
      true: {
        width: '100%',
      },
    },
    pill: {
      true: {
        borderRadius: '$pill',
      },
      false: {
        borderRadius: '$sm',
      },
    },
    isDark: {
      true: {},
      false: {},
    },
  },
});

export default StyledButton;
