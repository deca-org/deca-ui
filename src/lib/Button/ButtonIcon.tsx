import { CSS, styled } from '@lib/Theme/stitches.config';
import React from 'react';

const StyledButtonIcon = styled('span', {
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  height: '100%',
  '& svg': {
    background: 'transparent',
  },
  compoundVariants: [
    {
      isSingle: false,
      side: 'left',
      css: {
        mr: '$1',
      },
    },
    {
      isSingle: false,
      side: 'right',
      css: {
        ml: '$1',
      },
    },
    {
      isSingle: true,
      size: 'sm',
      css: {
        '& svg': {
          height: 'calc(100% - $0)',
          width: '100%',
        },
      },
    },
  ],
  variants: {
    size: {
      sm: {
        '& svg': {
          height: 'calc(100% - $1)',
          width: '100%',
        },
      },
      md: {
        '& svg': {
          height: 'calc(100% - $2)',
          width: '100%',
        },
      },
      lg: {
        '& svg': {
          height: 'calc(100% - $3)',
          width: '100%',
        },
      },
    },
    isSingle: {
      true: {
        '& svg': {
          height: 'calc(100% - $2)',
          width: '100%',
        },
      },
      false: {},
    },
    side: {
      left: {},
      right: {},
    },
  },
});

export interface ButtonIconProps {
  children: React.ReactNode | undefined;
  className?: string;
  css?: CSS;
  size?: 'sm' | 'md' | 'lg';
  isSingle?: boolean;
  side?: 'left' | 'right';
}

const ButtonIcon = ({
  children,
  className,
  css,
  size,
  isSingle,
  side,
  ...props
}: ButtonIconProps) => (
  <StyledButtonIcon
    className={className}
    css={css}
    size={size}
    isSingle={isSingle}
    side={side}
    {...props}
  >
    {children}
  </StyledButtonIcon>
);

export default ButtonIcon;
