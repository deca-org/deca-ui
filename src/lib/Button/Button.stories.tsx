import { DecaUIProvider } from '@lib/Theme';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { PurchaseTagAlt } from '@styled-icons/boxicons-solid/PurchaseTagAlt';
import React from 'react';

import Button from './Button';

export default {
  title: 'Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Default = Template.bind({});

Default.args = {
  children: 'Button',
  color: 'primary',
  disabled: false,
  css: {},
  className: '',
  maxWidth: false,
  variant: 'solid',
  role: 'button',
  size: 'md',
  pill: false,
};

export const IconButton = Template.bind({});
IconButton.args = {
  ...Default.args,
  children: undefined,
  icon: <PurchaseTagAlt />,
};

export const IconWithLabel = Template.bind({});
IconWithLabel.args = {
  ...Default.args,
  icon: <PurchaseTagAlt />,
};

export const RightSideIconWithLabel = Template.bind({});
RightSideIconWithLabel.args = {
  ...Default.args,
  iconRight: <PurchaseTagAlt />,
};

export const WithTheme = Template.bind({});

WithTheme.args = { ...Default.args };
WithTheme.decorators = [
  (Story) => (
    <DecaUIProvider
      theme={{
        colors: {
          primary: '$green600',
        },
        radii: {
          sm: '15px',
        },
      }}
    >
      <Story />
    </DecaUIProvider>
  ),
];

export const DarkMode = Template.bind({});

DarkMode.args = { ...Default.args };
DarkMode.parameters = { backgrounds: { default: 'dark' } };

DarkMode.decorators = [
  (Story) => {
    return (
      <DecaUIProvider mode="dark">
        <Story />
      </DecaUIProvider>
    );
  },
];
