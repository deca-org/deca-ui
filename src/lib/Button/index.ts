import Button from './Button';
export * from './Button';
export { Button };
export default Button;
