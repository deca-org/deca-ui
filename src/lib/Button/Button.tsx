import { ThemeContext } from '@lib/Theme';
import { CSS, StandardColors } from '@lib/Theme/stitches.config';
import {
  __DEV__,
  PolymorphicComponentPropWithRef,
  PolymorphicRef,
} from '@lib/Utils';
import clsx from 'clsx';
import React, { useMemo } from 'react';

import StyledButton from './Button.styles';
import ButtonIcon from './ButtonIcon';

/**
 * Buttons allow users to take actions, and make choices, with a single tap.
 */
interface Props {
  /**
   * The content of the component.
   */
  children?: React.ReactNode | undefined;
  /**
   * The role attribute of the component.
   */
  role?: string;
  /**
   * ClassName applied to the component.
   * @default ''
   */
  className?: string;
  /**
   * Disabled state applied to the component.
   * @default false
   */
  disabled?: boolean;
  /**
   * Function that executes when button is clicked.
   */
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  /**
   * Size of the component.
   * @default md
   */
  size?: 'sm' | 'md' | 'lg';
  /**
   * Override default CSS style.
   */
  css?: CSS;
  /**
   * The variant to use.
   * @default solid
   */
  variant?: 'solid' | 'outlined' | 'ghost' | 'flat';
  /**
   * Color to use.
   * @default primary
   */
  color?: StandardColors;
  /**
   * Set button width to 100%.
   * @default false
   */
  maxWidth?: boolean;
  /**
   * Icon on the left side of the component.
   */
  icon?: React.ReactNode;
  /**
   * Icon on the right side of the component.
   */
  iconRight?: React.ReactNode;
  /**
   * CSS applied to icon on the left side of the component.
   */
  iconLeftCss?: CSS;
  /**
   * CSS applied to icon on the right side of the component.
   */
  iconRightCss?: CSS;
  /**
   * Have button be pill shaped
   */
  pill?: boolean;
}

export type ButtonProps<T extends React.ElementType> =
  PolymorphicComponentPropWithRef<T, Props>;

export type ButtonComponent = (<C extends React.ElementType = 'button'>(
  props: ButtonProps<C>
) => React.ReactElement | null) & { displayName?: string };

export const Button: ButtonComponent = React.forwardRef(
  <T extends React.ElementType = 'button'>(
    {
      role = 'button',
      as,
      css,
      icon,
      iconRight,
      iconLeftCss,
      iconRightCss,
      onClick,
      color = 'primary',
      disabled = false,
      variant = 'solid',
      size = 'md',
      maxWidth = false,
      className = '',
      children,
      pill = false,
      ...btnProps
    }: ButtonProps<T>,
    ref?: PolymorphicRef<T>
  ) => {
    const hasText = useMemo(
      () => React.Children.count(children) !== 0,
      [children]
    );

    // ensures only icon or only iconRight is present
    const singleIcon = useMemo(
      () =>
        (!hasText && !iconRight && icon) || (!hasText && !icon && iconRight)
          ? true
          : false,
      [children, iconRight, icon]
    );

    if (disabled) {
      onClick = undefined;
    }

    const preClass = 'decaButton';

    const { dark } = React.useContext(ThemeContext);

    return (
      <StyledButton
        role={role}
        as={as}
        css={css}
        onClick={onClick}
        color={color}
        disabled={disabled}
        isDisabled={disabled}
        variant={variant}
        size={size}
        maxWidth={maxWidth}
        className={clsx(className, `${preClass}-root`)}
        ref={ref}
        singleIcon={singleIcon}
        pill={pill}
        isDark={dark}
        {...btnProps}
      >
        {(icon && hasText) || (icon && iconRight) ? (
          <ButtonIcon
            size={size}
            isSingle={false}
            side="left"
            css={iconLeftCss}
            className={`${preClass}-leftIcon`}
          >
            {icon}
          </ButtonIcon>
        ) : (
          icon && (
            <ButtonIcon
              size={size}
              isSingle={true}
              side="left"
              css={iconLeftCss}
              className={`${preClass}-leftIcon`}
            >
              {icon}
            </ButtonIcon>
          )
        )}
        {hasText && children}
        {(iconRight && hasText) || (icon && iconRight) ? (
          <ButtonIcon
            size={size}
            isSingle={false}
            side="right"
            css={iconRightCss}
            className={`${preClass}-rightIcon`}
          >
            {iconRight}
          </ButtonIcon>
        ) : (
          iconRight && (
            <ButtonIcon
              size={size}
              isSingle={true}
              side="right"
              css={iconRightCss}
              className={`${preClass}-rightIcon`}
            >
              {iconRight}
            </ButtonIcon>
          )
        )}
      </StyledButton>
    );
  }
);

if (__DEV__) {
  Button.displayName = 'DecaUI.Button';
}

export default Button;
