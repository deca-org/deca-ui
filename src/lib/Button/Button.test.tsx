import { ArrowCircleRightOutline } from '@styled-icons/evaicons-outline/ArrowCircleRightOutline';
import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/react';
import React from 'react';

import Button from './Button';

describe('components/Button', () => {
  it('renders content', () => {
    const { getByText } = render(<Button>Label</Button>);
    expect(getByText('Label')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Button>Label</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('onClick event fires', () => {
    const mockFn = jest.fn();
    const { getByText } = render(<Button onClick={mockFn}>Label</Button>);
    fireEvent.click(getByText('Label'));
    expect(mockFn.mock.calls.length).toBe(1);
  });

  it('renders all colors', () => {
    const { asFragment } = render(
      <>
        <Button color="primary">Label</Button>
        <Button color="secondary">Label</Button>
        <Button color="success">Label</Button>
        <Button color="warning">Label</Button>
        <Button color="error">Label</Button>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all sizes', () => {
    const { asFragment } = render(
      <>
        <Button size="sm">Label</Button>
        <Button size="md">Label</Button>
        <Button size="lg">Label</Button>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders all variants', () => {
    const { asFragment } = render(
      <>
        <Button variant="solid">Label</Button>
        <Button variant="outlined">Label</Button>
        <Button variant="ghost">Label</Button>
        <Button variant="flat">Label</Button>
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should ignore events when disabled', () => {
    const mockFn = jest.fn();
    const { getByText } = render(
      <Button disabled onClick={mockFn}>
        Label
      </Button>
    );
    fireEvent.click(getByText('Label'));
    expect(mockFn.mock.calls.length).toBe(0);
  });

  it('renders icon', () => {
    const { container, getByText } = render(
      <Button icon={<ArrowCircleRightOutline />}>Label</Button>
    );
    const iconEl = container.querySelector('svg');
    expect(iconEl).toBeInTheDocument();
    expect(getByText('Label')).toBeInTheDocument();
  });

  it('renders icon on right side', () => {
    const { container, getByText } = render(
      <Button iconRight={<ArrowCircleRightOutline />}>Label</Button>
    );
    const iconEl = container.querySelector('svg');
    expect(iconEl).toBeInTheDocument();
    expect(getByText('Label')).toBeInTheDocument();
  });
});
