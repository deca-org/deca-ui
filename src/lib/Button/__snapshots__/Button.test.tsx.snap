// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-ePLPad-cv decaButton-root"
    role="button"
  >
    Label
  </button>
</DocumentFragment>
`;

exports[`components/Button renders all colors 1`] = `
<DocumentFragment>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-ePLPad-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-bmmCCE-color-secondary c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-gCieqL-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-cbktrZ-color-success c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-kkCveC-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-fWqKPQ-color-warning c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-dQlwmf-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-cgBQIK-color-error c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-bAuLhl-cv decaButton-root"
    role="button"
  >
    Label
  </button>
</DocumentFragment>
`;

exports[`components/Button renders all sizes 1`] = `
<DocumentFragment>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-cmVdgU-size-sm c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-ePLPad-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-ePLPad-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-eIDFyj-size-lg c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-ePLPad-cv decaButton-root"
    role="button"
  >
    Label
  </button>
</DocumentFragment>
`;

exports[`components/Button renders all variants 1`] = `
<DocumentFragment>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-ePLPad-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-fdQTW-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-hSqWZv-cv decaButton-root"
    role="button"
  >
    Label
  </button>
  <button
    class="c-hpqThE c-hpqThE-jsOpSy-color-primary c-hpqThE-dcPSsq-size-md c-hpqThE-bDGmTT-isDisabled-false c-hpqThE-cycqzJ-pill-false c-hpqThE-kJLEgH-cv decaButton-root"
    role="button"
  >
    Label
  </button>
</DocumentFragment>
`;
